packages/@react-aria/datepicker/src/useDateSegment.ts

```diff
@@ -98,6 +98,7 @@ export function useDateSegment(segment: DateSegment, state: DateFieldState, ref:
       let parsed = parser.parse(newValue);
       if (newValue.length === 0 || parsed === 0) {
         state.clearSegment(segment.type);
+        focusManager.focusPrevious();
```

Comment:
Personally, I feel like focus should remain in the field. I'm used to deleting the entire segment to type in my new input, but open to other opinions from the team

Reply:
I think if the segment is already cleared, it's improved UX if backspace moves focus to the previous segment. We just should ensure the focus doesn't wrap (i.e. don't move focus if focus is already on the first segment.)

Reply:
Yeah, I think I agree with Daniel on this one. Personally, if I backspace to clear out a segment, that means I want to enter a new value for that same segment, not go back to the previous segment.

Reply:
Just sharing my thoughts here - in the current implementation of DateSegment, it automatically moves to the next field once the input is complete, but it doesn’t revert to the previous field when using the backspace key. This is aligned with Chrome’s behavior.

However, thinking about consistency, it somehow feels more intuitive if the backspace would actually take us back to the previous field. 

In the case of Safari, the focus doesn’t automatically move to the next field after input, nor does it revert to the previous one with a backspace. This can be confirmed at the following link.
https://developer.mozilla.org/docs/Web/HTML/Element/input/date

Reply:
If everyone is on board with this behavior, I will make the adjustment as suggested by @reidbarber, ensuring that the focus doesn't move back if it's already on the first segment.

Reply:
Feels like we are leaning towards the behavior @reidbarber described but I'll hopefully can get you a definitive answer tomorrow.

Reply:
Confirmed the behavior with the rest of the team, feel free to go ahead and implement what @reidbarber described!

Reply:
Okay, thanks! I'll give it a try.

Reply:
@LFDanLu 
After giving it some thought, I realized that in our case, since we're not using the wrap option with `focusManager.focusPrevious()`, it seems like we won't encounter any wrapping even if the focus is on the first element. This should align well with the behavior we're aiming for, as per @reidbarber's suggestion.

Reply:
Yep, that is correct. The only thing to change here would be to **only** focus the previous date segment if the user hits Backspace on a empty segment. This could be done by just first checking if the segment is already empty and then calling focusPrevious.

Reply:
I feel like our current implementation is fine. It seems more natural for the focus to shift to the previous input field when the final value is deleted, not just when the current segment is empty. For instance, if the focus is on a placeholder and backspace moves the focus back, that feels like too much.

Reply:
IMO I feel like auto moving focus once the input segment is cleared is too much because it assumes the user is wants to move to the previous date segment when they could just be trying to input another value in the original input segment. Auto moving focus forward when a segment is filled in on the other hand feels fine because there isn't another operation that the user might be doing once the segment is filled in

Reply:
I concur with @LFDanLu +1

Reply:
I've tried to fix it. please review when you have time!